docs/guides/composition-revisions-example.md

```diff
@@ -0,0 +1,302 @@
+metadata:
+  name: myvpcs.aws.example.upbound.io
+spec:
+  group: aws.example.upbound.io
+  names:
+    kind: MyVPC
+    plural: myvpcs
+  versions:
+  - name: v1alpha1
+    served: true 
+    referenceable: true 
+    schema:
+      openAPIV3Schema:
+        type: object 
+        properties:
+          spec:
+            type: object 
+            properties:
+              id:
+                type: string 
+                description: ID of this VPC that other objects will use to refer to it. 
+            required:
+            - id
+```
+
+## Verify the Composition revision
+Verify that Crossplane created the Composition revision
+```shell
+kubectl get compositionrevisions -o=custom-columns=NAME:.metadata.name,REVISION:.spec.revision,LABEL:.metadata.labels.channel
+```
+Expected Output:
+```shell
+NAME                                    REVISION   LABEL
+myvpcs.aws.example.upbound.io-ad265bc   1          dev
+```
+
+The label `dev` is automatically created from the Composition.
+
+## Create Composite Resources
+
+### Default update policy
+Create an XR without a `compositionUpdatePolicy` defined. The update policy is `Automatic` by default:
+```yaml
+apiVersion: aws.example.upbound.io/v1alpha1
+kind: MyVPC
+metadata:
+  name: vpc-auto
+spec:
+  id: vpc-auto
+```
+Expected Output:
+```shell
+myvpc.aws.example.upbound.io/vpc-auto created
+``` 
+
+### Manual update policy
+Create an Composite Resource with `compositionUpdatePolicy: Manual` and `compositionRevisionRef`.
+```yaml
+apiVersion: aws.example.upbound.io/v1alpha1
+kind: MyVPC
+metadata:
+  name: vpc-man
+spec:
+  id: vpc-man
+  compositionUpdatePolicy: Manual
+  compositionRevisionRef:
+    name: myvpcs.aws.example.upbound.io-ad265bc
+```
+
+Expected Output:
+```shell
+myvpc.aws.example.upbound.io/vpc-man created
+``` 
+
+
+### Using a selector
+Create an XR with a `compositionRevisionSelector` of `channel: dev`:
+```yaml
+apiVersion: aws.example.upbound.io/v1alpha1
+kind:  MyVPC
+metadata:
+  name: vpc-dev
+spec:
+  id: vpc-dev
+  compositionRevisionSelector:
+    matchLabels:
+      channel: dev
+```
+Expected Output:
+```shell
+myvpc.aws.example.upbound.io/vpc-dev created
+``` 
+
+Create an XR with a `compositionRevisionSelector` of `channel: staging`:
+```yaml
+apiVersion: aws.example.upbound.io/v1alpha1
+kind: MyVPC
+metadata:
+  name: vpc-staging
+spec:
+  id: vpc-staging
+  compositionRevisionSelector:
+    matchLabels:
+      channel: staging
+```
+
+Expected Output:
+```shell
+myvpc.aws.example.upbound.io/vpc-staging created
+``` 
+
+## Verify the Composite Resources
+Verify the Composite Resource with the label `channel: staging` doesn't have a `REVISION`.  
+All other XRs have a `REVISION` matching the created Composition Revision.
+```shell
+kubectl get composite -o=custom-columns=NAME:.metadata.name,SYNCED:.status.conditions[0].status,REVISION:.spec.compositionRevisionRef.name,POLICY:.spec.compositionUpdatePolicy,MATCHLABEL:.spec.compositionRevisionSelector.matchLabels
```

Comment:
Using zsh and this command failed as is: 

```
zsh: no matches found: -o=custom-columns=NAME:.metadata.name,SYNCED:.status.conditions[0].status,REVISION:.spec.compositionRevisionRef.name,POLICY:.spec.compositionUpdatePolicy,MATCHLABEL:.spec.compositionRevisionSelector.matchLabels
```
It worked fine, after I quote as follows:

```sh
kubectl get composite -o="custom-columns=NAME:.metadata.name,SYNCED:.status.conditions[0].status,REVISION:.spec.compositionRevisionRef.name,POLICY:.spec.compositionUpdatePolicy,MATCHLABEL:.spec.compositionRevisionSelector.matchLabels"
```